### PR TITLE
search users using search API

### PR DESCRIPTION
**Pull request by alexei-led**
support searching users with pagination

### CODE REVIEW

**Comment by david-doit-intl** [comment]
@eggman64, @Donnie, @christophstrasen please could you review. This would a very useful api to call from our application, otherwise we'd have to create another fork which is not desired.

Thank you for your efforts.

**Review by david-doit-intl — CHANGES_REQUESTED** [review]
Its probably not needed to change the gitignore file.

Can you please also sign your commits.

File: zendesk/search.go [hunk 1]
```diff
@@ -90,3 +100,29 @@ func (c *client) SearchTickets(term string, options *ListOptions, filters ...Fil
...
+		queryString = fmt.Sprintf(`%s /"%s/"`, queryString, term)
+	}
+	params.Set("query", queryString)
+	out := new(UserSearchResults)
+	err = c.get(fmt.Sprintf("/api/v2/search.json?%s", params.Encode()), out)
+	if err != nil {
+		return nil, err
+	}
+	return out, nil
+}
```
Comment (david-doit-intl) [inline zendesk/search.go:128]: please add a line ending here

**Comment by eggman64** [comment]
> Its probably not needed to change the gitignore file.
> 
> Can you please also sign your commits.

@alexei-led please remove the `.gitignore` change

**Comment by david-doit-intl** [comment]
@eggman64 thank you very much, we will get those changes sorted!

**Review by eggman64 — CHANGES_REQUESTED** [review]
Thanks so much for the contribution! Please add a test for the new method - we have good coverage and would love to keep it high :)
You can refer to any existing tests

**Comment by david-doit-intl** [comment]
@eggman64 test added

**Review by david-doit-intl — APPROVED** [review]

**Comment by alexei-led** [comment]
@eggman64 can you, please, review this PR and merge it if it's ok. Thank you!

**Review by eggman64 — APPROVED** [review]

**Pull request merged**